xlators/mount/fuse/src/fuse-bridge.c

```diff
@@ -4079,6 +4126,21 @@ fuse_setxattr_resume(fuse_state_t *state)
     state->fd = fd_lookup(state->loc.inode, state->finh->pid);
 #endif /* GF_TEST_FFOP */
 
+    if (dict_get_sizen(state->xattr, GF_NAMESPACE_KEY)) {
```

Comment:
Should this check be in removexattr as well? Shouldn't this check be present in gfapi part too?

Reply:
Haven't handled 'removexattr()' as removing namespace in a running process is tricky... ie, we have to recursively reset namespace pointer in whole inode tree under that namespace.

Reply:
my idea was, even if there is gfapi client, the namespace setting _always_ happens through a fuse client (that too a special one, ie, pid < 0). Hence haven't handled setxattr in gfapi.